core/src/main/scala/org/apache/spark/rdd/SqlNewHadoopRDD.scala

```diff
@@ -250,6 +258,19 @@ private[sql] class SqlNewHadoopRDD[K, V](
 }
 
 private[spark] object SqlNewHadoopRDD {
+
+  /**
+   * The thread variable for the name of the current file being read. This is used by
+   * the InputFileName function
+   */
+  private[this] val inputFileName: ThreadLocal[String] = new ThreadLocal[String]
```

Comment:
in order to optimize this, you can store UTF8String directly here, so we don't need to do the conversion every time.